src/test/java/pl/debuguj/parkingspacessystem/controllers/ParkingSpaceControllerTest.java

```diff
@@ -0,0 +1,356 @@
+package pl.debuguj.parkingspacessystem.controllers;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import pl.debuguj.parkingspacessystem.ParkingspacessystemApplication;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+import static org.junit.Assert.*;
+
+/**
+ * Created by grzesiek on 14.10.17.
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = ParkingspacessystemApplication.class)
+@WebAppConfiguration
+public class ParkingSpaceControllerTest {
+
+    @Autowired
+    private WebApplicationContext context;
+
+    private MockMvc mockMvc;
+
+    @Before
+    public void setup(){
+        this.mockMvc = MockMvcBuilders.webAppContextSetup(context).build();
+    }
+
+    @Test
+    public void shouldReturnCorrectHttpStatusAndPayload() throws Exception {
+        //GIVEN
+        String givenUrl = new StringBuilder()
```

Comment:
trzeba korzystać ze StringBuildera? mock MVC chyba daje takie możliwości

Reply:
W książkach z których korzystałem było to zrobione w ten sposób. Dobrze wiedzieć żei mockMvc daje takie możliwości .